File: pipeline/dose.py
from pipeline.config import BASELINE_C, WALK_SPEED_MPS


def dose_c_min(
    temp_c: float,
    len_m: float,
    baseline_c: float = BASELINE_C,
    walk_speed_mps: float = WALK_SPEED_MPS,
) -> float:
    return max(temp_c - baseline_c, 0.0) * (len_m / walk_speed_mps) / 60.0


def weight_cool(dose_value: float, len_m: float, lambda_detour: float) -> float:
    return dose_value + lambda_detour * len_m

File: web/src/lib/dose.ts
export const WALK_SPEED_MPS = 1.2

export function doseCMin(tempC: number, lenM: number, baselineC: number): number {
  return (Math.max(tempC - baselineC, 0) * (lenM / WALK_SPEED_MPS)) / 60
}

export function weightCool(doseValue: number, lenM: number, lambdaDetour: number): number {
  return doseValue + lambdaDetour * lenM
}

export function walkMinutes(lenM: number): number {
  return lenM / WALK_SPEED_MPS / 60
}
